src/app/qgsdiagramproperties.cpp

```diff
@@ -119,31 +122,52 @@ QgsDiagramProperties::QgsDiagramProperties( QgsVectorLayer *layer, QWidget *pare
     mDiagramFrame->setEnabled( false );
   }
 
-  //insert placement options
-  mPlacementComboBox->blockSignals( true );
-  switch ( layerType )
+  // set placement methods page based on geometry type
+
+  switch ( mLayer->geometryType() )
   {
     case QgsWkbTypes::PointGeometry:
-      mPlacementComboBox->addItem( tr( "Around Point" ), QgsDiagramLayerSettings::AroundPoint );
-      mPlacementComboBox->addItem( tr( "Over Point" ), QgsDiagramLayerSettings::OverPoint );
+      stackedPlacement->setCurrentWidget( pagePoint );
       mLinePlacementFrame->setVisible( false );
       break;
     case QgsWkbTypes::LineGeometry:
-      mPlacementComboBox->addItem( tr( "Around Line" ), QgsDiagramLayerSettings::Line );
-      mPlacementComboBox->addItem( tr( "Over Line" ), QgsDiagramLayerSettings::Horizontal );
+      stackedPlacement->setCurrentWidget( pageLine );
       mLinePlacementFrame->setVisible( true );
       break;
     case QgsWkbTypes::PolygonGeometry:
-      mPlacementComboBox->addItem( tr( "Around Centroid" ), QgsDiagramLayerSettings::AroundPoint );
-      mPlacementComboBox->addItem( tr( "Over Centroid" ), QgsDiagramLayerSettings::OverPoint );
-      mPlacementComboBox->addItem( tr( "Perimeter" ), QgsDiagramLayerSettings::Line );
-      mPlacementComboBox->addItem( tr( "Inside Polygon" ), QgsDiagramLayerSettings::Horizontal );
+      stackedPlacement->setCurrentWidget( pagePolygon );
       mLinePlacementFrame->setVisible( false );
       break;
-    default:
+    case QgsWkbTypes::NullGeometry:
       break;
+    case QgsWkbTypes::UnknownGeometry:
+      qFatal( "unknown geometry type unexpected" );
```

Comment:
1. Is this never ever expected and if it happens the reason should be investigated? If yes, use a Q_ASSERT.

2. Can this happen sometimes without any concequence at all? Do nothing (or use QgsDebugMsgLevel).

3. Does this happen sometimes and should be of interest for the user? If yes use a label or other UI hint or - depending on severity - use QgsMessageLog.

I guess it's 2.